extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesPortBuildItem.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.kubernetes.spi;
+
+import org.jboss.builder.item.SimpleBuildItem;
+
+public final class KubernetesPortBuildItem extends SimpleBuildItem {
+
+    private final int port;
+
+    public KubernetesPortBuildItem(int port) {
```

Comment:
Are these always *just* port numbers?  Is there any concept in Kube for multiple bind addresses?

Reply:
As far as the `container` is concerned these are just numbers.
However, these should be named. So I'd like to see at least a property called `name` or something.

Reply:
There is yes, you could have multiple ports. In such a case we would need to have a name for the them as well.
The reason I went with this simplistict approach is that Quarkus currently only binds to a single port. Would you like me to make `KubernetesPortBuildItem` a `MultiBuildItem` and also add the name so we account for future changes?

Reply:
The safest bet is to assume that any Quarkus service can bind to any number of addresses and ports.

Reply:
What would the name represent?  Would it be the "kind" of port e.g. `http` or would it be something else?

Reply:
Exactly, `http` is what I have used so far as the hardcoded name. So I guess I will make `KubernetesPortBuildItem` a `MultiBuildItem` and have the producer supply the name as well

Reply:
Perfect.  Is there a limit to the number of ports with the same name?

Reply:
Do we need to do something to account for multiple addresses?  Like for example would we need to say "only ports bound to this address get `KubernetesPortBuildItem` instances" or something like that?

Reply:
AFAIK, the names need to be unique, so we would need to make sure we add a suffix if necessary in the processor.

I don't think we need to take the address into consideration at this point. One thing that _might_ make sense would be to only consider non localhost addresses, but even that might not be correct in cases where the user wants to use a sidecar (which shares localhost). So I would say that unless we run into any weird unforeseen scenarios in the future, address shouldn't be needed.

Reply:
So then we will have to have some kind of check to ensure uniqueness.  If a user declares two HTTP ports or addresses, I guess that means they don't get automatic K8s integration (because they're doing something nonsensical for that use case) unless we add some kind of idiom around naming each bind address.

Reply:
Yes, makes sense